course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -0,0 +1,125 @@
+            expect(thing).not.toBeUndefined();
+            expect(thing).not.toBeNull();
+        });
+    });
+
+    describe('$scope setup', function() {
+        beforeEach(function() {
+            controller = $controller('PeopleController', {$scope: scope });
+        });
+
+        it('should set the course instance id', function() {
+            expect(scope.course_instance_id).toEqual($routeParams.course_instance_id);
+        });
+
+        it('should set dtColumns up', function() {
+            expect(scope.dtColumns).not.toBeUndefined();
+            expect(scope.dtColumns).not.toBeNull();
+        });
+
+        it('should set dtOptions up', function() {
+            expect(scope.dtOptions).not.toBeUndefined();
+            expect(scope.dtOptions).not.toBeNull();
+        });
+
+        it('should have a null dtInstance', function() {
+            expect(scope.dtInstance).toBeNull();
+        });
+    });
+
+    describe('setTitle', function() {
+        var ci;
+        beforeEach(function() {
+            ci = {
+                course_instance_id: $routeParams.course_instance_id,
+                title: 'Test Title',
+            };
+            courseInstances.instances = {};
+        });
+        afterEach(function() { courseInstances.instances = {}; });
+
+        it('should work when courseInstances has the course instance', function() {
+            courseInstances.instances[ci.course_instance_id] = ci;
+            controller = $controller('PeopleController', {$scope: scope });
+            expect(scope.title).toEqual(ci.title);
+        });
+
+        it('should work whenCourseInstances is empty', function() {
+            controller = $controller('PeopleController', {$scope: scope});
+            $httpBackend.expectGET('/angular/reverse/?djng_url_name=icommons_rest_api_proxy&djng_url_args=api%2Fcourse%2Fv2%2Fcourse_instances%2F1234567890%2F')
+                .respond(200, JSON.stringify(ci));
+            $httpBackend.flush();
+            expect(scope.title).toEqual(ci.title);
+        });
+    });
+
+    describe('dt cell render functions', function() {
+        beforeEach(function() {
+            controller = $controller('PeopleController', {$scope: scope});
+        });
+
+        it('renderName', function() {
+            var full = {
+                profile: {
+                    name_first: 'Joe',
+                    name_last: 'Student',
+                },
+            };
+            var result = scope.renderName(undefined, undefined, full, undefined);
+            expect(result).toBe('Student, Joe');
+        });
+
+        it('renderId', function() {
+            var full = {
+                profile: {role_type_cd: 'STUDENT'},
+                user_id: 123456,
+            }
+            var result = scope.renderId(undefined, undefined, full, undefined);
+            expect(result).toBe('<badge ng-cloak role="STUDENT"></badge> 123456');
```

Comment:
This is just testing that our render function is returning the correct string, right?  At no point have we tried to hang this off the dom.

But yes, for dom elements dynamically generated by other sources (ie. datatables), you have to manually [compile them](https://github.com/Harvard-University-iCommons/canvas_account_admin_tools/pull/26/files#diff-e045fa70e4df999f02b9ae1ebead90b8R91) yourself.

The really awesome part about that is that the compiled dom (ie. what angular replaces your custom tag with) isn't simply replaced...it's inserted as child elements of your custom tag.  That's why I'm searching for children [here](https://github.com/Harvard-University-iCommons/canvas_account_admin_tools/pull/26/files#diff-5f5c4f29557cabab2a819964f13191b3R41), instead of just checking the tag name.